packages/transport-dom/src/create.ts

```
@@ -168,9 +175,9 @@ export const createChannelTransport = ({
         ),
         new Promise<TransportMessage>((resolve, reject) => {
           pending.set(requestId, (tev: TransportEvent) => {
-            if (isTransportMessage(tev, requestId)) {
+            if (isTransportMessage(tev) && tev.requestId === requestId) {
               resolve(tev);
-            } else if (isTransportError(tev, requestId)) {
+            } else if (isTransportError(tev)) {
               reject(errorFromJson(tev.error, tev.metadata, new ConnectError('Unary failed')));
             } else {
               reject(ConnectError.from(tev));
```

behavior of the response callback is slightly changed.

id parameter of the type guard was removed, and the types were made simpler. the id parameter was used to reject events that did not contain a specific `requestId`, so to achieve the same behavior it should be checked deliberately.

but, anything activating this callback has got it from the pending map by its id, so it's not as critical as it might seem. this change is more permissive.

1. the same: a successful response resolving the request is confirmed to match `requestId`
2. changed: a serialized error rejecting the request is not confirmed to match `requestId`

case 2 should never happen, because the message listener shouldn't pick up this callback unless the event it's handling contains the appropriate `requestId`.

if it somehow did happen, this is still an improvement because:

- the request will fail immediately with that error instead of timing out,
- the error content will be properly deserialized instead of passed to the generic fallback rejection